magma/magma
# feat(agw): add sctp up/downstream sock to mme service config

## Summary

Closes #8828.

Informed by https://github.com/magma/magma/commit/7c54225801bcd75a8d4ab6c9bf4190100745372c

### Changes

- Removed unused sctp configurations from MME service config
  - in_streams (SCTP_INSTREAMS in config)
  - out_streams (SCTP_OUTSTREAMS in config)
- Adds sctp configurations for MME service config
  - SCTP_UPSTREAM_SOCK (default unix:///tmp/sctpd_upstream.sock)
  - SCTP_DOWNSTREAM_SOCK (default unix:///tmp/sctpd_downstream.sock)

## Test Plan

- Ran `lte integration test` to ensure no regression in nominal parameter
- Added unit test to verify configuration parsing of new fields
- In magma VM, ensured config change results in new unix domain socket path
  - vagrant ssh magma
  - cd `/magma/lte/gateway`
  - vim `/vagrant/configs/templates/mme.conf.template`
    - update `SCTP_UPSTREAM_SOCK = unix:///tmp/sctpd_upstream_test.sock`
    - update `SCTP_DOWNSTREAM_SOCK = unix:///tmp/sctpd_downstream_test.sock`
  - make run
  - vim `/var/opt/magma/tmp/mme.conf` (validate config)
    - validate that the template was filled correctly (should be no-op for new parameters)
  - ls -lh `/tmp/sctpd_*.sock`
    - should contain
      - sctpd_downstream.sock (generated by sctpd)
      - /tmp/sctpd_upstream_2.sock (generated by mme with new config)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>
